File: projects/health-wise-app/src/app/core/models/blood-pressure-reading.ts
import { Person } from '.';

export interface PersonBloodPressure {
  person: Person;
  bloodPressureReading: BloodPressureReading;
}

export interface BloodPressureReading {
  id: string;
  systole: number;
  diastole: number;
  heartRate: number;
  personId: string;
  dateTaken: string;
}

File: projects/health-wise-api/src/HealthWiseBackend.API/Core/Entities/BloodPressureReading.cs
using HealthWiseBackend.API.Enums;
using System;

namespace HealthWiseBackend.API.Entities
{
  public class BloodPressureReading
  {
    private BloodPressureReading()
    {

    }

    public BloodPressureReading(int systole, int diastole, int heartRate)
    {
      Systole = systole;
      Diastole = diastole;
      HeartRate = heartRate;
    }

    public BloodPressureReading(Guid id, int systole, int diastole, int heartRate): this(systole, diastole, heartRate)
    {
      Id = id;
    }

    public Guid Id { get; private set; }
    public int Systole { get; private set; }
    public int Diastole { get; private set; }
    public int HeartRate { get; private set; }
    public DateTime DateTaken { get; set; }

    public string PersonId { get; set; }
  }
}
